src/main/java/bio/terra/service/filedata/flight/ingest/IngestDriverStep.java

```diff
@@ -0,0 +1,279 @@
+                if (currentRunning < concurrentFiles) {
+                    // Compute how many loads to launch
+                    int launchCount = concurrentFiles - currentRunning;
+                    if (candidateCount < launchCount) {
+                        launchCount = candidateCount;
+                    }
+
+                    launchLoads(context, launchCount, candidates.getCandidateFiles(), profileId, loadId);
+                    currentRunning += launchCount;
+                }
+
+                // Wait until some loads complete
+                waitForAny(context, loadId, concurrentFiles, currentRunning);
+            }
+        } catch (DatabaseOperationException ex) {
+            return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, ex);
+        }
+
+        BulkLoadArrayResultModel result = makeLoadResult(loadId, context);
+        workingMap.put(JobMapKeys.RESPONSE.getKeyName(), result);
+
+        return StepResult.getStepResultSuccess();
+    }
+
+    @Override
+    public StepResult undoStep(FlightContext context) {
+        return StepResult.getStepResultSuccess();
+    }
+
+    private BulkLoadArrayResultModel makeLoadResult(UUID loadId, FlightContext context) {
+        // Get the summary stats and fill in our specific information
+        BulkLoadResultModel summary = loadService.makeBulkLoadResult(loadId);
+        summary.loadTag(loadTag).jobId(context.getFlightId());
+
+        // Get the file load results
+        List<BulkLoadFileResultModel> fileResults = loadService.makeBulkLoadFileArray(loadId);
+
+        return new BulkLoadArrayResultModel()
+            .loadSummary(summary)
+            .loadFileResults(fileResults);
+    }
+
+
+
+    private void waitForAny(FlightContext context, UUID loadId, int concurrentLoads, int originallyRunning)
+        throws DatabaseOperationException {
+        while (true) {
+            waiting();
+            LoadCandidates candidates = getLoadCandidates(context, loadId, concurrentLoads);
+            if (candidates.getRunningLoads().size() < originallyRunning) {
+                break;
+            }
+        }
+    }
+
+    private void waitForAll(FlightContext context, UUID loadId, int concurrentLoads) throws DatabaseOperationException {
+        while (true) {
+            waiting();
+            LoadCandidates candidates = getLoadCandidates(context, loadId, concurrentLoads);
+            if (candidates.getRunningLoads().size() == 0) {
+                break;
+            }
+        }
+    }
+
+    private void waiting() {
+        logger.debug("Waiting for file loads to complete...");
+
+        // TODO: make a settable for the sleep timer for this wait
+        int waitSeconds = 60;
+        try {
+            TimeUnit.SECONDS.sleep(waitSeconds);
+        } catch (InterruptedException iex) {
+            Thread.currentThread().interrupt();
+            throw new FileSystemExecutionException("Bulk ingest driver interruped!");
```

Comment:
Let's include `iex` along with the message here

Reply:
Done